### ISSUE TITLE
Error when doing predict() with previous dtwclust object

### ISSUE THREAD

**Issue by sergiowang** [asardaes/dtwclust#11]
Hi Alex. It has been a while since I last used your package. Today I found that the predict() function doesn't work with my previously saved dtwclust object. When I use `predict(old_object,series)`, it returned `Error in (function (x, centroids = NULL, ...)  : 
  could not find function "check_parallel"` while it works well with new dtwclust object that I created today. I think this problem is different from the issue I created before: #9 .So I would like to know if there is any method to solve this problem without creating a new dtwclust object.

**Comment by asardaes** [comment]
Hello. Indeed some of the internal functions changed or were removed, so `check_parallel` is no longer used. It can be fixed, although it depends on the specific arguments you used to create `old_object`.

It occurs to me that you could do the following and it might work, allowing you to use predict (you'll need at least version 3.1.1 from CRAN):

```r
updated_object <- as(old_object, "TSClusters")
```

This also depends on which version of the package you had when you created `old_object`. If some slots are missing, it may fail, but try it and let me know.

**Comment by sergiowang** [comment]
I've tried your solution and it fixes the problem perfectly. Really appreciate your help !

**Comment by sergiowang** [comment]
Hi Alex. Yesterday I updated the dtwclust package, and met another problem with the previous saved dtwclust object again.
There are two previous saved dtwclus objects, when I do `new_ob<-as(old_ob,"TSClusters")`, one object (let it be `ob1`) is successfully coerced while another (`ob2`) is failed and returns:
```Error in match.arg(allcent, c("fcm", "fcmdd")) : 
  'arg' should be one of fcm? fcmdd?``` 
I think this may results from the original call when I created these objects.
`ob1@call` is `dtwclust(data = train_list, type = "partitional", k = 3, distance = "dtw", 
    centroid = "mean", seed = 100)`
`ob2@call` is `dtwclust(data = train_list, type = "fuzzy", k = 3, distance = "dtw", 
    centroid = "mean", seed = 100)`
 Any suggestions about how to solve this problem?
